msteams-platform/bots/how-to/conversations/channel-messages-with-rsc.md

```diff
@@ -12,15 +12,35 @@ The resource-specific consent (RSC) permissions model, originally developed for
 
 ## Enable bots to receive all channel or chat messages
 
-> [!NOTE]
->
-> The ability for bots to receive all messages in chats using `ChatMessage.Read.Chat` is available only in [public developer preview for Teams](../../../resources/dev-preview/developer-preview-intro.md).
-
 The `ChannelMessage.Read.Group` and `ChatMessage.Read.Chat` RSC permissions are being extended to bots. With user consent and app installation, these permissions:
 
 * Allow a specified graph application to get all messages in channels and chats, respectively.
 * Enable a bot defined in the app manifest to receive all conversations messages without being @mentioned in relevant contexts where the permissions apply.
 
+```csharp
+
+// When rsc is enabled the method will be called even when bot is addressed without being @mentioned.
+// This code snippet will allow the bot to ignore all messages that do not @mention the bot.
+protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+{
+        // Ignore the message if bot was not mentioned.
+        if (!turnContext.Activity.GetMentions().Any(mention => mention.Mentioned.Id.Equals(turnContext.Activity.Recipient.Id, StringComparison.OrdinalIgnoreCase)))
+        {
+            return;
+        }
+
+        // Sends an activity to the sender of the incoming activity.
+        await turnContext.SendActivityAsync(MessageFactory.Text("Using RSC the bot can receive messages across channels or chats in team without being @mentioned."));
+}
+
+```
+
+> [!NOTE]
+>
+> * The ability for bots to receive all messages in chats using `ChatMessage.Read.Chat` is available only in [public developer preview for Teams](../../../resources/dev-preview/developer-preview-intro.md) and will only be enabled after a re-installation or new installation into a chat. 
+> * Once enabled, the bot will continue to receive all messages even when the client switches out of public developer preview. 
+> * If you have an app that is currently using the `ChatMessage.Read.Chat` RSC permission for Graph scenarios, then you should test the app following the [guide](#chat-messages) below and modify the app if needed.
```

Comment:
I think this should be in the "Important" section below.
Also I'd make this a stronger call to action:
- test the app and make modifications before we make the feature generally available on XXX date
- otherwise their bot will receive all messages for new installations
- if they don't want this behavior, they should modify their code by filtering out messages that don't at-mention the bot

Reply:
Thanks for the review @aosolis. Should we give a specific date for GA? I'm wondering if there could be confusion if we were to miss the GA date due to unknown factors.

Reply:
I added a link to the feature roadmap instead of giving a date. I think this should be clear enough and avoids confusion in the future if dates change.